src/java.base/share/classes/jdk/internal/misc/OperatingSystem.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.misc;
+
+import jdk.internal.util.StaticProperty;
+import jdk.internal.vm.annotation.ForceInline;
+
+import java.util.Objects;
+
+/**
+ * Enumeration of operating system types and testing for the current OS.
+ * The enumeration can be used to dispatch to OS specific code or values.
+ * Checking if a specific operating system is current uses a simple
+ * static method for each operating system.
+ * <p>
+ * For example,
+ * {@snippet lang = "java":
+ * if (OperatingSystem.isWindows()) {
+ *     // Windows only code.
+ * } else if (OperatingSystem.isLinux()) {
+ *     // Linux only code
+ * }
+ *}
+ *
+ * Alternatively, compare with the {@linkplain #current() current} operating system.
+ * For example,
+ * {@snippet lang = "java":
+ * if (OperatingSystem.current() == Windows) {
```

Comment:
Doesn't `Windows` need to be prefixed with `OperatingSystem` here? Ditto for dispatch example following.

Reply:
Yes, either a static import or a qualified name is needed.
(This is internal javadoc).